src/arns.lua

```diff
@@ -378,6 +378,53 @@ function arns.getActiveArNSNamesBetweenTimestamps(startTimestamp, endTimestamp)
 	return activeNames
 end
 
+function arns.getTotalArNSNamesBetweenTimestampsUnsafe(startTimestamp, endTimestamp)
+	local totalActiveNames = 0
+	local totalGracePeriodNames = 0
+	local records = arns.getRecordsUnsafe()
+
+	for _, record in pairs(records) do
+		if record.type == "permabuy" then
+			totalActiveNames = totalActiveNames + 1
+		elseif record.type == "lease" and record.endTimestamp then
+			if record.startTimestamp <= startTimestamp and record.endTimestamp >= endTimestamp then
+				totalActiveNames = totalActiveNames + 1
+			elseif record.endTimestamp + constants.gracePeriodMs >= endTimestamp then
+				totalGracePeriodNames = totalGracePeriodNames + 1
+			end
+		end
+	end
+
+	return {
+		totalActiveNames = totalActiveNames,
+		totalGracePeriodNames = totalGracePeriodNames,
+	}
+end
```

Comment:
apologies, my comment wasn't clear. i was suggesting scoping one function for grace period arns names, we already have one for active

```suggestion
function arns.getGracePeriodArNSNamesBetweenTimestamps(startTimestamp, endTimestamp)
	local records = arns.getRecordsUnsafe()
	local gracePeriodNames = {}
	for name, record in pairs(records) do
		if record.endTimestamp + constants.gracePeriodMs >= endTimestamp then
			table.insert(gracePeriodNames, name)
		end
	end
	return gracePeriodNames
end

function arns.getTotalArNSNamesBetweenTimestampsUnsafe(startTimestamp, endTimestamp)
        local activeNames = arns.getActiveArNSNamesBetweenTimestamps(startTimestamp, endTimestamp),
        local gracePeriodNames = arns.getGracePeriodArNSNamesBetweenTimestamps(startTimestamp, endTimestamp)
	return {
		totalActiveNames = #totalActiveNames,
		totalGracePeriodNames = #totalGracePeriodNames,
	}
end
```